neptune_optuna/impl/__init__.py

```diff
@@ -336,150 +429,205 @@ def load_study_from_run(run: neptune.Run):
+                    vis.plot_slice(study, target=target, target_name=target_name)
+                )
+
+            if log_plot_intermediate_values and any(
+                trial.intermediate_values for trial in study.trials
+            ):
+                # Intermediate values plot if available only if the above condition is met
+                temp_handle["plot_intermediate_values"] = neptune.types.File.as_html(
+                    vis.plot_intermediate_values(
+                        study, target=target, target_name=target_name
+                    )
+                )
+
+            if log_plot_optimization_history:
+                temp_handle["plot_optimization_history"] = neptune.types.File.as_html(
+                    vis.plot_optimization_history(
+                        study, target=target, target_name=target_name
+                    )
+                )
+
+    if (
+        vis.is_available
+        and log_plot_pareto_front
+        and study._is_multi_objective()
+        and visualization_backend == "plotly"
+    ):
+        handle["plot_pareto_front"] = neptune.types.File.as_html(
+            vis.plot_pareto_front(study, target_names=namespaces)
+        )
+
+
+def _log_single_trial(
+    run, study: optuna.Study, trial: optuna.trial.FrozenTrial, namespaces, best=False
+):
+    handle = run["best"] if best else run["trials"]
+
+    handle[f"trials/{trial._trial_id}/datetime_start"] = trial.datetime_start
+    handle[f"trials/{trial._trial_id}/datetime_complete"] = trial.datetime_complete
+    handle[f"trials/{trial._trial_id}/duration"] = trial.duration
+    handle[f"trials/{trial._trial_id}/distributions"] = trial.distributions
+    handle[f"trials/{trial._trial_id}/intermediate_values"] = trial.intermediate_values
+    handle[f"trials/{trial._trial_id}/params"] = trial.params
+
+    if study._is_multi_objective():
+        handle[f"trials/{trial._trial_id}/values"] = {
+            f"{namespaces[k]}": v for k, v in enumerate(trial.values)
+        }
 
+    else:
+        handle[f"trials/{trial._trial_id}/value"] = trial.value
 
-def _log_trials(run, trials: Iterable[optuna.trial.FrozenTrial]):
-    handle = run['trials']
+        if best:
+            handle["value"] = trial.value
+            handle["param"] = trial.params
+            handle["value|param"] = f"value: {trial.value}| params: {trial.params}"
+        else:
+            handle["values"].log(trial.value, step=trial._trial_id)
+            handle["params"].log(trial.params)
+            handle["values|params"].log(f"value: {trial.value}| params: {trial.params}")
+
+    if trial.state.is_finished() and trial.state != optuna.trial.TrialState.COMPLETE:
+        handle[f"trials/{trial._trial_id}/state"] = repr(trial.state)
+
+
+def _log_trials(
+    run,
+    study: optuna.Study,
+    trials: Iterable[optuna.trial.FrozenTrial],
+    namespaces,
+    best=False,
+):
+    if not study.best_trials and best:
```

Comment:
`return None` as this function now executes logging instead of returning a dictionary :wink: